src/main/java/org/spongepowered/api/text/selector/Selector.java

```diff
@@ -24,57 +24,78 @@
  */
 package org.spongepowered.api.text.selector;
 
+import com.google.common.base.Optional;
 import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.extent.Extent;
 
 import java.util.List;
-import java.util.Map;
-
-import javax.annotation.Nullable;
 
 /**
- * Represents an immutable selector. A selector starts with the {@code @} symbol
- * and is followed by a single character signifying which selector is being used
- * (known as the type), and finally the optional arguments in brackets. For
- * example, the all player selector is {@code @a}, and with a radius of 20 it
- * would be {@code @a[r=20]}.
+ * Represents an immutable selector of targets, as used in commands.
+ *
+ * <p>In Vanilla, selectors are mostly represented as plain text, starting with
+ * an {@code @} symbol and followed by a single character signifying the type,
+ * and finally the (optional) arguments in brackets.</p> <p>As an example, the
+ * all player selector is {@code @a}, and with a radius of 20 it would be
+ * {@code @a[r=20]}.</p>
  *
  * @see <a href="http://minecraft.gamepedia.com/Selector#Target_selectors">
- *      TargetSelectors on the Minecraft Wiki</a>
+ *      Target selectors on the Minecraft Wiki</a>
  */
 public interface Selector {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (61/60 (101%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:53994 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.84)
	(envelope-from <noreply@github.com>)
	id 1YNeYG-0003Um-Df
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:34:44 +0000
Date: Tue, 17 Feb 2015 01:34:44 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1424165684;
	bh=tgCgeXr/EG123LpfZWNfzDMEtBUmPRKpEhzggoKVNq8=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=exoh9kOtSZ/UqNXpf3nOKzKfJpRR6NgPLqf3kh009c8BxIDlHY2LrsnQX2vAvduui
	 dIUaau2rcJicl135Go5gXqGMjl30TZcO6hZeTg+59SAw+pgY60gSd0r/HJVax2dd4R
	 plz4zhyDfi/gyRGm2eQe6+d6sEdPGQtGIfbX+iIE=
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803009@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30b3450d66_32883fb39ed2729c1426985";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bf4adbfd4983dca98a840af52ca528f5af1b87af4292cf0000000110facd3492a170ce017a76c1@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAvwQ3_X_803EKD4dsDBsjsqeVDieGks5nswK0gaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk


----==_mimepart_54e30b3450d66_32883fb39ed2729c1426985
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>   *
>   * @see <a href="http://minecraft.gamepedia.com/Selector#Target_selectors">
> - *      TargetSelectors on the Minecraft Wiki</a>
> + *      Target selectors on the Minecraft Wiki</a>
>   */
>  public interface Selector {

Could also be a class

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803009
----==_mimepart_54e30b3450d66_32883fb39ed2729c1426985
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803009">src/main/java/org/spongepowered/api/text/selector/Selector.java</a>:</p>
<pre style='color:#555'>&gt;   *
&gt;   * @see &lt;a href=&quot;http://minecraft.gamepedia.com/Selector#Target_selectors&quot;&gt;
&gt; - *      TargetSelectors on the Minecraft Wiki&lt;/a&gt;
&gt; + *      Target selectors on the Minecraft Wiki&lt;/a&gt;
&gt;   */
&gt;  public interface Selector {
</pre>
<p>Could also be a class</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803009">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv4HcListYQRMQeI0C3h41bbaXaGzks5nswK0gaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803009"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30b3450d66_32883fb39ed2729c1426985--